### PR TITLE
testdisk: update to 7.1

### PR DESCRIPTION
**Pull request by ylluminarious**
#### Description

Updates `testdisk` to the latest stable release, [version 7.1](https://www.cgsecurity.org/wiki/TestDisk_7.1_Release), which was [released July 7, 2019](https://www.cgsecurity.org/wiki/TestDisk_Download).

I also removed a patch for `libewf` which no longer seems necessary in the newer version.

###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on
macOS 10.15.7 19H2
Xcode 12.2 12B45b

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [x] checked your Portfile with `port lint`?
- [ ] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [x] tested basic functionality of all binary files?

### CODE REVIEW

File: sysutils/testdisk/Portfile [hunk 1]
```diff
@@ -30,16 +29,13 @@ depends_lib     port:libcomerr \
...
-
 use_bzip2       yes
 
 homepage        http://www.cgsecurity.org/wiki/TestDisk
 master_sites    http://www.cgsecurity.org/
 
-checksums           rmd160  56ff652a9ddf1c2b5cf96493e9aed8e7f7942d1f \
-                    sha256  00bb3b6b22e6aba88580eeb887037aef026968c21a87b5f906c6652cbee3442d
+checksums           rmd160  481f3ec24e3d2ed3b15a8efb038abf88db9bc55d \
+                    sha256  1413c47569e48c5b22653b943d48136cb228abcbd6f03da109c4df63382190fe
```
Comment (cjones051073) [inline sysutils/testdisk/Portfile:38]: Please add the size field to the list of checksums

**Review by cjones051073 — CHANGES_REQUESTED** [review]

File: sysutils/testdisk/Portfile (hunk 1 above)
Comment (ylluminarious) [reply sysutils/testdisk/Portfile]: @cjones051073 This should be working now.

**Review by cjones051073 — APPROVED** [review]

**Comment by cjones051073** [comment]
Can you please check the build failures for 10.15 ?

**Comment by ylluminarious** [comment]
@cjones051073 From a quick glance through the log files, it appears this may have something to do with `osxfuse`. `testdisk` depends on two library ports which in turn depend on `osxfuse`: `ntfs-3g` and `libewf`.  It's possible that [since `osxfuse` doesn't work on Catalina](https://trac.macports.org/ticket/59316), this might also be what is causing the build failure on 10.15. I worked around this by already having `osxfuse` installed (an older version) and those two libraries on my system. I temporarily commented out those lines to get it to build.

**Comment by ryandesign** [comment]
> It's possible that [since `osxfuse` doesn't work on Catalina](https://trac.macports.org/ticket/59316), this might also be what is causing the build failure on 10.15.

Yup, that's what the log says:

```
Installing dependency (17 of 24) 'osxfuse' with variants '' (requesting '') ... [FAIL]
```

**Pull request merged**